tensorflow/python/distribute/mirrored_strategy.py

```diff
@@ -515,14 +516,11 @@ def _experimental_make_numpy_dataset(self, numpy_input, session):
 
   def _distribute_datasets_from_function(self, dataset_fn,
                                                       options):
-    if options.replication_mode == distribute_lib.InputReplicationMode.PER_REPLICA:
+    if options.experimental_replication_mode == distribute_lib.InputReplicationMode.PER_REPLICA:
       self._input_workers_devices = (
           tuple((device_util.canonicalize("/device:CPU:0", d), (d,)) for d in self._devices))
```

Comment:
actually can we move this logic to `self._input_workers_with_options` method? since that has the `options` as well, it can do entire check, instead of checking some things here and some things there?

Also, regarding setting it to `self._input_workers_devices`, my overall opinion is that we should just not set any value to that anymore, since it can change based on which option was used in the last call to `strategy.distribute_dataset...`. for now, if it's simpler for you, it's fine to update it in this case. (just move all this logic to `_input_workers_with_options` so it is isolate in one place.

Reply:
logic moved per discussion.